### PR TITLE
Added loglogistic cdf and rng

### PR DESCRIPTION
**Pull request by gregorp90**
## Summary

Implemented `loglogistic_cdf` with manual gradients and `loglogistic_rng` [#2320](https://github.com/stan-dev/math/issues/2320#issuecomment-781315580). The latter is based on sampling from Uniform(0, 1) and transforming the samples with the inverse cdf.

## Tests

Added a cdf to distribution tests and a rng test to unit tests.

## Release notes

Implementation of the loglogistic cumulative distribution function and random number generator.

## Checklist

- [x] Math issue #2320

- [x] Copyright holder: Gregor Pirš

### CODE REVIEW

**Comment by gregorp90** [comment]
@SteveBronder @t4c1 Hi, would anyone be up for reviewing this? Or recommend another reviewer? Thanks!

**Review by t4c1 — CHANGES_REQUESTED** [review]
cdf looks more or less good. I did not work much with rngs, so I would prefer if somebody else reviews that part. Maybe @bbbales2?

**Comment by bbbales2** [comment]
The rng implementation and tests look correct to me.

File: stan/math/prim/prob/loglogistic_cdf.hpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    const auto& cdf_div_elt = to_ref_if<!is_constant_all<T_y>::value
+                                            + !is_constant_all<T_scale>::value
+                                            + !is_constant_all<T_shape>::value
+                                        >= 2>(cdf / prod_all);
+    if (!is_constant_all<T_y, T_scale>::value) {
+      const auto& alpha_div_times_beta = to_ref_if<
+          !is_constant_all<T_y>::value + !is_constant_all<T_scale>::value == 2>(
+          alpha_div_y_pow_beta * beta_val);
+      if (!is_constant_all<T_y>::value) {
+        const auto& y_deriv = alpha_div_times_beta * inv(y_val) * prod_all_sq;
```
[the remark this answers is not in the document]
Comment (gregorp90) [reply stan/math/prim/prob/loglogistic_cdf.hpp:105]: Thank you for the review! I will combine `y_deriv` with the next expression.

I have a question regarding your suggested change. I've been using the `inv` function here to guard against problems if both of the terms (`alpha_div_times_beta` and `y_val`) are integers. Or do we not need this, as these values are already converted to double or var somewhere along the line, maybe when we declare `y_ref`? On the other hand, we never come to this calculation if at least one of them is not var, so is that the reason why we can assume that one of these terms is not an integer? Just trying to fully understand the reasoning here so I can write better implementations next time.

[the remark this answers is not in the document]
Comment (t4c1) [reply stan/math/prim/prob/loglogistic_cdf.hpp:105]: > On the other hand, we never come to this calculation if at least one of them is not var, so is that the reason why we can assume that one of these terms is not an integer?

That is fine, although it is non-trivial reasoning so maybe a comment in the code would be helpful. 

In other cases you can use `promote_scalar<double>(arg)` to convert one of the arguments to double and avoid integer division.

[the remark this answers is not in the document]
Comment (gregorp90) [reply stan/math/prim/prob/loglogistic_cdf.hpp]: Great, thank you for the explanation! I will change the code according to your suggestions.

**Review by t4c1 — APPROVED** [review]

**Pull request merged**